tower-http/CHANGELOG.md

```diff
@@ -7,7 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 # Unreleased
 
-None.
+- Add `metrics::Traffic` for measuring how many responses or errors a service is
+  producing.
```

Comment:
This is potentially a more philosophical thought but `Traffic` feels like it should almost do all of this, _plus_ track response latency, _plus_ track the in-flight count.  The last major thing to track, IMO, is response latency... which would bring the middleware count up to three.

Reply:
Yeah I've had similar thoughts. I've considered calling it something like `RequestLifecycleCallbacks` 😜 but better.

The intended use case is to use this together with some metrics backend chosen by the user. At Embark we use the `metrics` crate and prometheus meaning all the aggregation is taken care of.

As mentioned I also do want to provide another crate that provides ready to use integration between tower-http and the metrics crate. However I don't think baking the metrics crate directly in to tower-http is appropriate.

Reply:
@tobz Do you think think `RequestResponseCallbacks` would be a better name?

Reply:
Hmmm, something with lifecycle in the name definitely sounds better.  Just spitballing, but maybe like... lifecycle hooks? Something like that.

Reply:
I've renamed and moved some things around. This middleware is now called `LifeCycleHooks` and lives in `tower_http::life_cycle`, so doesn't live in `metrics` anymore. I agree I think this makes more sense.

@tobz what do you think?